src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -158,19 +162,250 @@ private void testMapAllTypes(boolean registerFactory) throws SQLException {
+      assertEquals(Time.valueOf(LocalTime.of(15, 39, 20)), resultArray[1]);
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapTimestampArray() throws SQLException {
+    try (ResultSet resultSet =
+        resultSetFn.apply(
+            "SELECT ARRAY_CONSTRUCT(TO_TIMESTAMP_NTZ('2021-12-23 09:44:44'), TO_TIMESTAMP_NTZ('2021-12-24 09:55:55'))::ARRAY(TIMESTAMP)"); ) {
+      Timestamp[] resultArray = (Timestamp[]) resultSet.getArray(1).getArray();
+      assertEquals(Timestamp.valueOf(LocalDateTime.of(2021, 12, 23, 10, 44, 44)), resultArray[0]);
+      assertEquals(Timestamp.valueOf(LocalDateTime.of(2021, 12, 24, 10, 55, 55)), resultArray[1]);
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapBooleanArray() throws SQLException {
+    try (ResultSet resultSet =
+        resultSetFn.apply("SELECT ARRAY_CONSTRUCT(true,false)::ARRAY(BOOLEAN)"); ) {
+      Boolean[] resultArray = (Boolean[]) resultSet.getArray(1).getArray();
+      assertEquals(true, resultArray[0]);
+      assertEquals(false, resultArray[1]);
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapBinaryArray() throws SQLException {
+    try (ResultSet resultSet =
+        resultSetFn.apply(
+            "SELECT ARRAY_CONSTRUCT(TO_BINARY('616263', 'HEX'),TO_BINARY('616263', 'HEX'))::ARRAY(BINARY)"); ) {
+      Object[] resultArray = (Object[]) resultSet.getArray(1).getArray();
+      assertArrayEquals(new byte[] {'a', 'b', 'c'}, (byte[]) resultArray[0]);
+      assertArrayEquals(new byte[] {'a', 'b', 'c'}, (byte[]) resultArray[1]);
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapArrayOfStructToMap() throws SQLException {
+    try (ResultSet resultSet =
+        resultSetFn.apply(
+            "SELECT ARRAY_CONSTRUCT({'x': 'abc', 'y': 1}, {'x': 'def', 'y': 2} )::ARRAY(OBJECT(x VARCHAR, y INTEGER))"); ) {
+      Map[] resultArray = (Map[]) resultSet.getArray(1).getArray();
+      assertEquals("{x=abc, y=1}", resultArray[0].toString());
+      assertEquals("{x=def, y=2}", resultArray[1].toString());
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapArrayOfArrays() throws SQLException {
+    try (ResultSet resultSet =
+        resultSetFn.apply(
+            "SELECT ARRAY_CONSTRUCT(ARRAY_CONSTRUCT({'x': 'abc', 'y': 1}, {'x': 'def', 'y': 2}) )::ARRAY(ARRAY(OBJECT(x VARCHAR, y INTEGER)))"); ) {
+      Map[][] resultArray = (Map[][]) resultSet.getArray(1).getArray();
+      assertEquals("{x=abc, y=1}", resultArray[0][0].toString());
+      assertEquals("{x=def, y=2}", resultArray[0][1].toString());
+    }
+  }
+
+  Function<String, ResultSet> resultSetFn =
```

Comment:
What is the purpose of this function? Maybe `firstRowOf` would be a better name?

Reply:
Fixed